File: 3rd Sem/OOP Lab/2022_10_11/q3.cpp
#include<iostream>
using namespace std;

class Circle{
    protected:
    int rad;
    public:
    Circle(int a){
        rad = a;
    }
    void area_cir(){
        float area = 3.14 * rad* rad;
        cout<<"Area: "<<area<<endl;
    }
};

class Rectangle: public Circle{
    protected:
    int bred;
    public:
    Rectangle(int len, int bre): Circle(len){
        bred = bre;
    }
    void area_rect(){
        int area = rad * bred;
        cout<<"Area: "<<area<<endl;
    }
};

class Cylinder: public Circle{
    int height;
    public:
    Cylinder(int radi, int hei): Circle(radi){
        height = hei;
    }
    void vol_cy(){
        float vol = 3.14 * rad * rad * height;
        cout<<"Volume: "<<vol;
    }
};

int main(){
    Rectangle r1(5,6);
    Cylinder c1(8,9);

    r1.area_cir();
    r1.area_rect();
    c1.area_cir();
    c1.vol_cy();

    return 0;
}

File: 4th sem/WT Lab/2023_02_27/q3.java
class data{
    int hei, bre;
    void get_data(int hei, int bre){
        this.hei = hei;
        this.bre = bre;
    }
    void put_data(){
        System.out.println("Height: "+ hei);
        System.out.println("Breadth: "+ bre);
    }
}

class rectangle extends data{
    void area(){
        System.out.println("Area: " + hei * bre);
    }
    void perimeter(){
        System.out.println("Perimeter: "+2*(hei + bre));
    }
}

class triangle extends data{
    void area(){
        System.out.println("Area: "+ (0.5 * bre * hei) );
    }
    void perimeter(){
        System.out.println("Perimeter: "+ (9 + bre+hei));
    }
}

public class q3 {
    public static void main(String[] args){
        triangle t1 = new triangle();
        rectangle r1 = new rectangle();

        t1.get_data(10, 20);
        r1.get_data(15, 25);

        t1.area();
        t1.perimeter();
        r1.area();
        r1.perimeter();
    }
}
